graylog2-web-interface/src/components/common/Scratchpad.jsx

```diff
@@ -0,0 +1,181 @@
+import React, { useContext, useState, useEffect, useRef } from 'react';
+import styled, { css } from 'styled-components';
+import { rgba } from 'polished';
+import ClipboardJS from 'clipboard';
+
+import teinte from 'theme/teinte';
+import Interactable from 'components/common/Interactable';
+import { Alert, Button, MenuItem, SplitButton } from 'components/graylog';
+import { BootstrapModalConfirm } from 'components/bootstrap';
+import Icon from 'components/common/Icon';
+import { ScratchpadContext } from 'routing/context/ScratchpadProvider';
+
+const LOCALSTORAGE_ITEM = 'gl-scratchpad';
+const DEFAULT_SCRATCHDATA = '';
+const TEXTAREA_ID = 'scratchpad-text-content';
+
+const ContentArea = styled.div`
+  display: flex;
+  flex-direction: column;
+  height: 100%;
+`;
+
+const Description = styled.p`
+  color: ${teinte.primary.due};
+  margin: 9px 0 6px;
+`;
+
+const Textarea = styled.textarea(props => css`
+  width: 100%;
+  resize: none;
+  flex: 1;
+  margin-bottom: 15px;
+  border: 1px solid ${props.copied ? teinte.tertiary.tre : teinte.secondary.tre};
+  box-shadow: inset 0 1px 1px rgba(0, 0, 0, .075),
+              0 0 8px ${rgba(props.copied ? teinte.tertiary.tre : teinte.secondary.tre, 0.6)};
+  transition: border 150ms ease-in-out, box-shadow 150ms ease-in-out;
+
+  :focus {
+    border-color: ${teinte.tertiary.due};
+    outline: none;
+  }
+`);
+
+const StyledAlert = styled(Alert)`
+  && {
+    padding: 6px 12px;
+    margin-bottom: 9px;
+    display: flex;
+    align-items: center;
+  }
+`;
+
+const AlertNote = styled.em`
+  margin-left: 6px;
+  flex: 1;
+`;
+
+const Footer = styled.footer`
+  display: flex;
+  justify-content: flex-end;
+  padding-bottom: 9px;
+`;
+
+const Scratchpad = () => {
+  let clipboard;
+  const storage = JSON.parse(localStorage.getItem(LOCALSTORAGE_ITEM)) || {};
```

Comment:
In case it makes sense, [`Store.js`](https://github.com/Graylog2/graylog2-server/blob/master/graylog2-web-interface/src/logic/local-storage/Store.js) is already taking care of some of the things in here.